db/dbtest/util.go

```diff
@@ -0,0 +1,71 @@
+package dbtest
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	dbm "github.com/cosmos/cosmos-sdk/db"
+)
+
+func AssertValid(t *testing.T, itr dbm.Iterator, expected bool) {
+	valid := itr.Valid()
+	require.Equal(t, expected, valid)
+}
+
+func AssertNext(t *testing.T, itr dbm.Iterator, expected bool) {
+	itr.Next()
+	// assert.NoError(t, err) TODO: look at fixing this
+	valid := itr.Valid()
+	require.Equal(t, expected, valid)
+}
+
+func AssertNextPanics(t *testing.T, itr dbm.Iterator) {
+	assert.Panics(t, func() { itr.Next() }, "checkNextPanics expected an error but didn't")
+}
+
+func AssertDomain(t *testing.T, itr dbm.Iterator, start, end []byte) {
+	ds, de := itr.Domain()
+	assert.Equal(t, start, ds, "checkDomain domain start incorrect")
+	assert.Equal(t, end, de, "checkDomain domain end incorrect")
+}
+
+func AssertItem(t *testing.T, itr dbm.Iterator, key []byte, value []byte) {
+	v := itr.Value()
+
+	k := itr.Key()
+
+	assert.Exactly(t, key, k)
+	assert.Exactly(t, value, v)
+}
+
+func AssertInvalid(t *testing.T, itr dbm.Iterator) {
+	AssertValid(t, itr, false)
+	AssertKeyPanics(t, itr)
+	AssertValuePanics(t, itr)
+	AssertNextPanics(t, itr)
+}
+
+func AssertKeyPanics(t *testing.T, itr dbm.Iterator) {
+	assert.Panics(t, func() { itr.Key() }, "checkKeyPanics expected panic but didn't")
+}
+
+func AssertValue(t *testing.T, db dbm.DBReader, key []byte, valueWanted []byte) {
+	valueGot, err := db.Get(key)
+	assert.NoError(t, err)
+	assert.Equal(t, valueWanted, valueGot)
+}
+
+func AssertValuePanics(t *testing.T, itr dbm.Iterator) {
+	assert.Panics(t, func() { itr.Value() })
+}
+
+func CleanupDBDir(dir, name string) {
+	err := os.RemoveAll(filepath.Join(dir, name) + ".db")
+	if err != nil {
+		panic(err)
+	}
```

Comment:
`require.Error` ?

Reply:
This function is unused actually, will remove